src/v/coproc/script_context_backend.cc

```diff
@@ -340,11 +340,11 @@ write_materialized(output_write_inputs replies, output_write_args args) {
     bool err{false};
     co_await ss::parallel_for_each(
       grs,
-      ss::coroutine::lambda(
         [args, &err](grouping_t::value_type& vt) -> ss::future<> {
+                return process_reply_group(
+                  vt.first, std::move(vt.second), args).handle_exception([args, &err](const std::exception_ptr& e) {
```

Comment:
nit: It would be more code churn, but something like this is probably more idiomatic?
```c++
          return process_reply_group(vt.first, std::move(vt.second), args)
            .handle_exception_type([](const bad_reply_exception& ex) {
                vlog(coproclog.error, "No source for reply: {}", ex);
            })
            .handle_exception_type(
              [](const script_failed_exception& ex) { throw ex; })
            .handle_exception_type(
              [id{args.id}](const malformed_batch_exception& ex) {
                  throw engine_protocol_failure(id, ex.what());
              })
              // ....
```